base/rational.jl

```diff
@@ -288,21 +288,34 @@ trunc{T}(::Type{T}, x::Rational) = convert(T,div(x.num,x.den))
 floor{T}(::Type{T}, x::Rational) = convert(T,fld(x.num,x.den))
 ceil{ T}(::Type{T}, x::Rational) = convert(T,cld(x.num,x.den))
 
-function round{T}(::Type{T}, x::Rational, ::RoundingMode{:Nearest})
-    q,r = divrem(x.num,x.den)
-    s = abs(r) < (x.den+one(x.den)+iseven(q))>>1 ? q : q+copysign(one(q),x.num)
-    convert(T,s)
+
+function round{Tf, Tr}(::Type{Tf}, x::Rational{Tr}, ::RoundingMode{:Nearest})
+    q,r = divrem(num(x), den(x))
+    s = q
+    if abs(r) >= abs((den(x)-copysign(Tr(4), num(x))+one(Tr)+iseven(q))>>1 + copysign(Tr(2), num(x)))
```

Comment:
That is actually a good question. I can `{Tf, Tr<Integer}` and add something like
```julia
function round{Tf, Tb::Bool}(::Type{Tf}, x::Rational)
convert(Tf, x)
end
```
Since `one(Bool) == true` and `zero(Bool) == false`, so I think the conversion should make sense. You're also making me think of another edge case here. If `den(x) == 0` no matter `Tr`, the rational is well defined, but `divrem` will fail, although `round` should return `-Inf` or `Inf` depending on the sign of `num(x)`.